File: Lab_3/app/src/main/java/com/example/bmi_app/model/QuestionItem.java
package com.example.bmi_app.model;


import java.util.ArrayList;
import java.util.Collections;

public class QuestionItem implements Cloneable{
    private String question;
    private int quizPicture;
    private String correctAnswer;
    private int correctAnswerId;
    private ArrayList <String> answers;

    public QuestionItem(
            String question,
            int picture,
            String correctAn,
            String wrongAn1,
            String wrongAn2,
            String wrongAn3)
    {
        this.answers = new ArrayList<>();
        this.question = question;
        this.quizPicture = picture;
        this.correctAnswer = correctAn;
        this.correctAnswerId = 0;
        this.answers.add(correctAn);
        this.answers.add(wrongAn1);
        this.answers.add(wrongAn2);
        this.answers.add(wrongAn3);
    }

    public String getQuestion() {
        return question;
    }
    public String getCorrectAnswer() {
        return answers.get(correctAnswerId);
    }
    public int getPhoto() { return quizPicture; }
    public String getAnswerById(int id){ return this.answers.get(id); }
    public void ShuffleMyAnswersForDisplay() {
        Collections.shuffle(answers);
        correctAnswerId = answers.indexOf(correctAnswer);
    }


    @Override
    protected QuestionItem clone() throws CloneNotSupportedException {
        QuestionItem clone = null;
        try
        {
            clone = (QuestionItem) super.clone();
        }
        catch (CloneNotSupportedException e)
        {
            throw new RuntimeException(e);
        }
        return clone;
    }


}

File: Lab_4_Lab_5/app/src/main/java/com/example/bmi_app/model/QuestionItem.kt
package com.example.bmi_app.model

import java.util.*

class QuestionItem(
    var question: String,
    picture: Int,
    correctAn: String,
    wrongAn1: String,
    wrongAn2: String,
    wrongAn3: String
) : Cloneable {

    var quizPicture:Int = picture
    var correctAnswer:String = correctAn
    var correctAnswerId:Int
    var answers = arrayListOf<String>()

    init {
        this.answers = arrayListOf<String>()
        this.correctAnswerId = 0
        this.answers.add(correctAn)
        this.answers.add(wrongAn1)
        this.answers.add(wrongAn2)
        this.answers.add(wrongAn3)
    }

    fun getQuest():String{
        return question
    }

    fun getCorrectAn():String {
        return answers[correctAnswerId]
    }

    fun getPhoto():Int{
    return quizPicture
}

    fun getAnswerById(id:Int):String{
        return this.answers.get(id)
    }

    fun ShuffleMyAnswersForDisplay(){
        answers.shuffle()
        correctAnswerId = answers.indexOf(correctAnswer)
    }

    @Throws(CloneNotSupportedException::class)
    public override fun clone(): QuestionItem {

        var clone:QuestionItem? = null
        try
        {
            clone = super.clone() as QuestionItem
        }
        catch (e: CloneNotSupportedException )
        {
            throw RuntimeException(e)
        }
        return clone
    }

}